### PR TITLE
modifications for mint amount distribution

### PR DESCRIPTION
**Pull request by antstalepresh**

### CODE REVIEW

File: x/mint/keeper/keeper.go [hunk 1]
```diff
@@ -181,7 +179,7 @@ func (k Keeper) AddCollectedFees(ctx sdk.Context, fees sdk.Coins) error {
 	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
 
 	// allocate pool allocation ratio to pool-incentives module account account
-	coins := sdk.NewCoins(k.GetPoolAllocatableAsset(ctx))
+	coins := sdk.NewCoins(k.GetPoolAllocatableAsset(ctx, fees))
```
Comment (sunnya97) [inline x/mint/keeper/keeper.go:182]: This shouldn't be called fees.

File: x/mint/keeper/keeper_test.go [hunk 2]
```diff
@@ -46,17 +46,10 @@ func (suite *KeeperTestSuite) TestGetPoolAllocatableAsset() {
...
-		suite.ctx,
-		suite.app.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName),
-		sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(1481290)), sdk.NewCoin("test", sdk.NewInt(12389190))),
-	)
-	suite.NoError(err)
-
-	coin = mintKeeper.GetPoolAllocatableAsset(suite.ctx)
-	suite.Equal("316258stake", coin.String())
+	coin = mintKeeper.GetPoolAllocatableAsset(suite.ctx, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(1481290)), sdk.NewCoin("test", sdk.NewInt(12389190))))
+	suite.Equal("296258stake", coin.String())
```
Comment (ValarDragon) [inline x/mint/keeper/keeper_test.go:54]: Can we pick some nicer numbers, or comment the choice of numbers / their relations? I have no idea what the % ratio between these numbers is by just looking at it

File: x/mint/keeper/keeper.go [hunk 3]
```diff
@@ -181,7 +179,7 @@ func (k Keeper) AddCollectedFees(ctx sdk.Context, fees sdk.Coins) error {
 	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
```
Comment (sunnya97) [inline x/mint/keeper/keeper.go:179]: This function should be called DistributeMintedCoins instead.

File: x/mint/keeper/keeper.go [hunk 4]
```diff
@@ -181,7 +179,7 @@ func (k Keeper) AddCollectedFees(ctx sdk.Context, fees sdk.Coins) error {
...
 
 	// allocate pool allocation ratio to pool-incentives module account account
-	coins := sdk.NewCoins(k.GetPoolAllocatableAsset(ctx))
+	coins := sdk.NewCoins(k.GetPoolAllocatableAsset(ctx, fees))
 	err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, k.feeCollectorName, poolincentivestypes.ModuleName, coins)
```
Comment (sunnya97) [inline x/mint/keeper/keeper.go:183]: These coins shouldn't go to the fee collector first.  This function should send the proper amount to fee collector bucket, and poolincentives type.

File: x/mint/keeper/keeper.go [hunk 5]
```diff
@@ -165,14 +165,12 @@ func (k Keeper) MintCoins(ctx sdk.Context, newCoins sdk.Coins) error {
...
-
-	feeCollector := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName)
-	asset := k.bankKeeper.GetBalance(ctx, feeCollector.GetAddress(), params.MintDenom)
-
-	return sdk.NewCoin(asset.Denom, asset.Amount.ToDec().Mul(params.PoolAllocationRatio).TruncateInt())
+	amount := fees.AmountOf(params.MintDenom)
+	return sdk.NewCoin(params.MintDenom, amount.ToDec().Mul(params.PoolAllocationRatio).TruncateInt())
 }
 
 // AddCollectedFees implements an alias call to the underlying supply keeper's
```
Comment (sunnya97) [inline x/mint/keeper/keeper.go:176]: This function should be called DistributeMintedCoins instead.

File: x/mint/keeper/keeper.go (hunk 3 above)
Comment (sunnya97) [inline x/mint/keeper/keeper.go:179]: Should be renamed to `minted_coins`.

Amount should be `minted_coins` * stakingProportion

File: x/mint/keeper/keeper_test.go (hunk 2 above)
Comment (antstalepresh) [reply x/mint/keeper/keeper_test.go]: Ok, it's nicer now as logic is simplified

**Pull request merged**